### PR TITLE
pageserver/http: remove status code boilerplate from swagger spec

### PR DESCRIPTION
**Pull request by VladLazar**
## Problem
We specify a bunch of possible error codes in the pageserver api swagger spec. This is error prone and annoying to work with. https://github.com/neondatabase/cloud/pull/11907 introduced generic error handling on the control plane side, so we can now clean up the spec.

## Summary of changes
* Remove generic error codes from swagger spec
* Update a couple route handlers which would previously return an error without a `msg` field in the response body.

Tested via https://github.com/neondatabase/cloud/pull/12340

Related https://github.com/neondatabase/cloud/issues/7238

## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by jcsp** [comment]
>Tested via https://github.com/neondatabase/cloud/pull/12340

Thanks for doing that.

@mtyazici if the control plane compiles and e2e tests pass, does that give us a good certainty that the removed error codes won't cause issues?  I'm thinking especially of the 409/412 cases that might be explicitly used in some logic somewhere.

File: pageserver/src/http/openapi_spec.yml [hunk 1]
```diff
@@ -93,90 +75,18 @@ paths:
...
-              schema:
-                $ref: "#/components/schemas/ForbiddenError"
         "404":
-          description: Tenant not found
+          description: Tenant not found. This is the success path.
           content:
             application/json:
               schema:
                 $ref: "#/components/schemas/NotFoundError"
-        "409":
```
Comment (jcsp) [inline pageserver/src/http/openapi_spec.yml:156]: For the 409/412 cases, we might want to leave them in: these do serve a useful purpose of encouraging a reader to realize that this endpoint can fail in ways that not all endpoints can.

**Comment by VladLazar** [comment]
> > Tested via [neondatabase/cloud#12340](https://github.com/neondatabase/cloud/pull/12340)
> 
> Thanks for doing that.
> 
> @mtyazici if the control plane compiles and e2e tests pass, does that give us a good certainty that the removed error codes won't cause issues? I'm thinking especially of the 409/412 cases that might be explicitly used in some logic somewhere.

There's some failures on https://github.com/neondatabase/cloud/pull/12340. Trying to disentangle test flakiness from the change.

File: pageserver/src/http/openapi_spec.yml (hunk 1 above)
Comment (mtyazici) [reply pageserver/src/http/openapi_spec.yml:156]: Agree, if the error may require special handling on our side it would be best to put it into the spec.

**Comment by mtyazici** [comment]
>@mtyazici if the control plane compiles and e2e tests pass, does that give us a good certainty that the removed error codes won't cause issues? I'm thinking especially of the 409/412 cases that might be explicitly used in some logic somewhere.

I now see we retry on any error when checking for `404` on resource deletion instead of `409/412`. Could be better to fix that logic as well if it's better.

>There's some failures on https://github.com/neondatabase/cloud/pull/12340. Trying to disentangle test flakiness from the change.

Failures seem to be related to how PR is titled, the tests are passing which shows there is no actual problem.

**Comment by VladLazar** [comment]
> Failures seem to be related to how PR is titled, the tests are passing which shows there is no actual problem.

I retriggered the tests and they passed. Likely just flakes.

File: pageserver/src/http/openapi_spec.yml (hunk 1 above)
Comment (VladLazar) [reply pageserver/src/http/openapi_spec.yml]: I did check the cplane code for special handling of 409 and 412 and there was none, but makes sense from a readability pov. Pushed the branch.

**Comment by VladLazar** [comment]
Tests are passing on https://github.com/neondatabase/cloud/pull/12340. I think we should be ok to merge.

**Review by jcsp — APPROVED** [review]

**Pull request merged**